### ISSUE TITLE
Don't break union all by name into two lines

### ISSUE THREAD

**Issue by aersam** [tconbeer/sqlfmt#611]
**Describe the bug**
If I use `union all by name` sqlfmt formats this as `union all\nby name` which is a bit annoying

**To Reproduce**
```python
import sqlfmt.api

print(sqlfmt.api.format_string("select * from foo union all by name select * from bar", mode=sqlfmt.api.Mode()))

```

**Expected behavior**
Should output:
```
select * 
from foo 
union all by name  
select * 
from bar 
```

**Actual behavior**
```
select * 
from foo 
union all
by name  
select * 
from bar 
```

**Additional context**
What is the output of `sqlfmt --version`? `0.21.4`

**Comment by tconbeer** [comment]
TIL about `union all by name`. Thank you

**Comment by tconbeer** [comment]
just shipped a patch in 0.22.0 -- thanks for the report!
